Source/SpeechToTextV1/WebSockets/SpeechToTextSocket.swift

```diff
@@ -192,22 +191,17 @@ internal class SpeechToTextSocket: WebSocketDelegate {
     }
 
     private func onErrorMessage(error: String) {
-        let error = RestError.failure(0, error)
+        let error = RestError.other(message: error)
         onError?(error)
     }
 
     private func isAuthenticationFailure(error: Error) -> Bool {
         if let error = error as? WSError {
             let matchesCode = (error.code == 400)
-            let matchesDescription = (error.message == "Invalid HTTP upgrade")
-            return matchesCode && matchesDescription
+            let matchesType     = (error.type == .upgradeError)
+            return matchesCode && matchesType
         }
-
-        let error = error as NSError
-        let matchesDomain = (error.domain == "WebSocket")
-        let matchesCode = (error.code == 400)
-        let matchesDescription = (error.localizedDescription == "Invalid HTTP upgrade")
-        return matchesDomain && matchesCode && matchesDescription
+        return false
```

Comment:
@mkistler Can you check this? I am confused about what the code here is doing. I thought that it was checking for an `NSError` that was being created on line 77 (when `connectAttempts` exceeded `maxConnectAttempts`), but it didn't looks like it would ever work since it expects the `error.localizedDescription` to equal "Invalid HTTP upgrade", which is missing the "Check credentials?" part of the `NSError` string.

Reply:
I can't explain this code.  It looks like Glenn Fisher originally wrote it over two years ago:
```
da2ef0f83       (Glenn R. Fisher        2016-09-11 23:28:04 -0500       205)        let matchesDescription = (description == "Invalid HTTP upgrade")
```

Reply:
In that case, I'm going to leave my changes as-is, and see if it causes any issues in the future.